influxql/call_iterator.go

```diff
@@ -495,6 +500,163 @@ func IntegerMedianReduceSlice(a []IntegerPoint) []FloatPoint {
 	return []FloatPoint{{Time: ZeroTime, Value: float64(a[len(a)/2].Value)}}
 }
 
+// newModeIterator returns an iterator for operating on a mode() call.
+func NewModeIterator(input Iterator, opt IteratorOptions) (Iterator, error) {
+
+	switch input := input.(type) {
+	case FloatIterator:
+		createFn := func() (FloatPointAggregator, FloatPointEmitter) {
+			fn := NewFloatSliceFuncReducer(FloatModeReduceSlice)
```

Comment:
The slice func reducers are really slow and I don't think it's advisable to use them for mode since mode doesn't actually need that information. First, be sure to try this with your benchmarks to see the difference, but I would imagine that creating a custom reducer type similar to the `MeanReducer` in `functions.go` and doing something like this would be better:

```
type FloatModeReducer struct {
    m map[float64]int
}

func (r *FloatModeReducer) AggregateFloat(p *FloatPoint) {
    m[p.Value]++
}

func (r *FloatModeReducer) Emit() []FloatPoint {
    value := 0
    max := 0
    for v, count := range r.m {
        if count < max || (count == max && v < value) {
            continue
        }
        value = v
    }
}
```

I would guess that this probably ends up being faster than the slice reducer just because of much fewer memory allocations. Can you try it out?